src/hera/variable.py

```diff
@@ -0,0 +1,32 @@
+from argo_workflows.models import (
+    IoArgoprojWorkflowV1alpha1Inputs,
+    IoArgoprojWorkflowV1alpha1Parameter,
+)
+from pydantic import BaseModel
+
+from hera.env import EnvSpec
+
+
+class VariableAsEnv(BaseModel):
```

Comment:
While not needed for this PR, I am wondering what do you think about naming this `TaskVariable`/`TaskVariableAsEnv`? Looking through the Argo docs it seems that we can expose many variable types. For instance, if Hera will support `Step` in the future, or global workflow variables, which I assume may have their respective properties to construct the required string `{{...}}`. Perhaps this is a bridge we'll cross at that point as this does not require over-engineering but really curious what you think! 🙂

Reply:
I could be wrong, but my understanding is that the VariableAsEnv part is the generic part. What the task and step expose is the specific, ie in the future, a `Step` may have a property `ip` which will return `"{{{{steps.<STEPNAME>.ip}}}}`. And a step will still be able to take a `VariableAsEnv`.